### PR TITLE
Setup DCO workflow

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does / why we need it**:

This PR adds a workflow to check DCO.
I added this workflow because the app to check DCO is now down.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

**Comment by t-kikuc** [comment]
Thanks!
https://github.com/dcoapp/app/issues/211#issuecomment-2148058964 is related.

After the DCO recovers, this action will be removed, right? @Warashi @khanhtc1202

**Comment by Warashi** [comment]
> After the DCO recovers, this action will be removed, right? @Warashi @khanhtc1202

I think so.
I noticed some actions in this workflow is not allowed in this repository, so I'll push some fix.

File: .github/workflows/dco.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: DCO
```
Comment (t-kikuc) [inline .github/workflows/dco.yaml:1]: [IMO] How about specifying this DCO is a workaround?

```suggestion
name: DCO-workaround
```

**Review by t-kikuc — COMMENTED** [review]
Thank you so much, I commented a nit.

**Comment by Warashi** [comment]
This workflow runs as this.
https://github.com/pipe-cd/pipecd/actions/runs/9377057788/job/25817994492?pr=4944

@khanhtc1202 Since I cannot modify branch protection rules, please change the required checks.

**Comment by khanhtc1202** [comment]
@Warashi Updated the branch protection rules to remove DCO workflow as required step 👍

**Review by khanhtc1202 — APPROVED** [review]
Thank you 👍

**Review by ffjlabo — APPROVED** [review]
Thanks!

**Comment by Warashi** [comment]
@khanhtc1202 
I merged this PR, so please update the branch protection rules to add DCO-workaround as required.

**Pull request merged**